### PR TITLE
re-write setup.sh's help message to use the new style

### PR DESCRIPTION
**Pull request by aendeavor**
# Description

This PR introduces the new style of writing usage information in this project to the `setup.sh` script.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project (my code **is** the new guideline :D)
- [x] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or ENVIRONMENT.md or the Wiki)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Here are the looks

2021-02-17-180924_1917x1051_scrot

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -97,72 +97,104 @@ function _inspect
...
-
-  help: Show this help dialogue
-
+  echo -e "\e[35mSETUP\e[31m(\e[93m1\e[31m)
+
+\e[38;5;214mNAME\e[39m
+    ${SCRIPT:-${0}} - docker-mailserver administration script
+
+\e[38;5;214mSYNOPSIS\e[39m
+    ./${SCRIPT:-${0}} [ OPTIONS\e[31m...\e[39m ] OBJECT COMMAND [ help \e[31m|\e[39m ARGUMENTS\e[31m...\e[39m ]
```
Comment (wernerfred) [inline setup.sh:106]: According to your screenshot it is:
`./setup.sh [OPTIONS] OBJECT COMMAND [help | ARGUMENTS]` which I find misleading as ongoing `OBJECTS` are elements of `COMMANDS` so this:
`./setup.sh [OPTIONS] COMMAND [help | ARGUMENTS]` seems more valid to me.

File: setup.sh [hunk 2]
```diff
@@ -97,72 +97,104 @@ function _inspect
...
+        ${0} alias add <EMAIL ADDRESS> <RECIPIENT>
+        ${0} alias del <EMAIL ADDRESS> <RECIPIENT>
+        ${0} alias list
+
+    \e[94mOBJECT\e[39m quota \e[31m:=\e[39m
+        ${0} quota set <EMAIL ADDRESS> [<QUOTA>]
+        ${0} quota del <EMAIL ADDRESS>
+
+    \e[94mOBJECT\e[39m config \e[31m:=\e[39m
+        ${0} config dkim [ ARGUMENTS\e[31m...\e[39m ]
```
Comment (wernerfred) [inline setup.sh:167]: Here I am missing an explanation which `ARGUMENTS` are possible. Or was your intention to offload this to the usage function of the called script?

File: setup.sh [hunk 3]
```diff
@@ -97,72 +97,104 @@ function _inspect
...
+        Add the email account \e[37mtest@domain.tld\e[39m. You will be prompted to input a
+        password afterwards since no password was supplied.
+
+    \e[37m./setup.sh config dkim size 2048 selector ds\e[39m
+        Creates keys of length 2048 bit in a default setup where domains are obtained from
+        your accounts. The DKIM selector used is 'ds'.
+
+    \e[37m./setup.sh config dkim help\e[39m
+        This will provide you with a detailed explanation on how to use the \e[37mconfig
+        dkim\e[39m command, showing what arguments can be passed and what they do.
```
Comment (wernerfred) [inline setup.sh:193]: I really appreciate the examples section 👍🏻 
Although dkim is using a selector which is not (yet) implemented

File: setup.sh (hunk 1 above)
Comment (aendeavor) [reply setup.sh:106]: I have altered it to now use an easier to understand schema. I have added a `SUBCOMMAND` after the `COMMAND` definition in this line:

``` BASH
COMMAND := { email | alias | quota | config | relay | debug } SUBCOMMAND
```

which makes all clear I think.

File: setup.sh (hunk 2 above)
Comment (aendeavor) [reply setup.sh:167]: Yes. I would not like to state them here, but only indicate that there are arguments. The reason being is that if all commands have args, this overview will become confusing.

File: setup.sh (hunk 3 above)
Comment (aendeavor) [reply setup.sh:193]: I've changed that to use the `domain ...` argument :D

**Pull request merged**